packages/react-form/src/hooks/field/test/field.test.tsx

```diff
@@ -762,6 +762,18 @@ describe('asChoiceField', () => {
   it('replaces value with checked', () => {
     expect(asChoiceField({value: true} as any)).toMatchObject({checked: true});
   });
```

Comment:
Since this is modifying an existing hook, we need to be careful about breakage. It looks like this suite didn't have a test that it works properly when _boolean_ values don't match. 

What would you think about that as a regression test? It's a bit redundant with your "projects an unchecked value from a predicate" test, but all the same it would probably be valuable since any consumers would be using it for boolean values to ensure there's no breakage.